Docs: Update root and CLI READMEs for v3-alpha (#194)
xmtp/libxmtp: examples/cli/README.md

> **Important**  
> This software is in **alpha** status and ready for you to start experimenting with. Expect frequent changes as we add features and iterate based on feedback.

## Send a double ratchet message

Use the CLI to send a [double ratchet message](https://github.com/xmtp/libxmtp/blob/main/README.md#double-ratchet-messaging) between test wallets on the XMTP `dev` network. 

1. Go to the `examples/cli` directory.

2. Create a sender wallet account (user1). Create an [installation key bundle](https://github.com/xmtp/libxmtp/blob/main/README.md#installation-key-bundles) and store it in the database. Grant the installation key bundle permission to message on behalf of the sender address. This will allow the CLI to message on behalf of the sender address.

   ```bash
   ./xli.sh --db user1.db3 register
   ```

3. Likewise, create a recipient wallet account (user2) and an installation key bundle.

   ```bash
   ./xli.sh --db user2.db3 register
   ```

4. Get the recipient's wallet address.

   ```bash
   ./xli.sh --db user2.db3 info
   ```

> 5. Send a message into the conversation. The message is sent using [one session between each installation](https://github.com/xmtp/libxmtp/blob/main/README.md#installation-key-bundles) associated with the sender and recipient. The message is encrypted using a session key dedicated to this single message per the [double ratchet algorithm](https://github.com/xmtp/libxmtp/blob/main/README.md#double-ratchet-messaging).

Reviewer: ```suggestion
5. Send a message into the conversation. The message is sent using [one session between each installation](https://github.com/xmtp/libxmtp/blob/main/README.md#installation-key-bundles) associated with the sender and recipient. The message is encrypted using a per-message encryption key derived using the [double ratchet algorithm](https://github.com/xmtp/libxmtp/blob/main/README.md#double-ratchet-messaging).
```

Just want to be careful using the term `session key` - in Signal's [spec](https://signal.org/docs/specifications/doubleratchet/), `session key` refers to the initial root key of the double ratchet algorithm, and `message key` refers to the encryption key used per-message.

   ```bash
   ./xli.sh --db user1.db3 send <user2_address> "hello"
   ```

6. List conversations.

   ```bash
   ./xli.sh --db user1.db3 list-conversations
   ```

> If you want to run the CLI against localhost, go to the root directory and run `dev/up`. Then run the CLI commands using the `--local` flag.

Author: @richardhuaaa Did I capture this info correctly? =) Just making sure...

> If you want to run the CLI against localhost, go to the root directory and run `dev/up`. Then run the CLI commands using the `--local` flag.

Reviewer: ```suggestion
If you want to run the CLI against localhost, go to the root directory and run `dev/up` to start a local server. Then run the CLI commands using the `--local` flag.
```

# CLI

![Status](https://img.shields.io/badge/Project_status-Alpha-orange)

This is a demo XMTP v3-alpha console client (CLI) that you can use to send and receive messages via the command line. Specifically, you can use it to try out [double ratchet messaging](#double-ratchet-messaging) and [installation key bundles](#installation-key-bundles) enabled by XMTP v3-alpha.

> **Important**  
> This software is in **alpha** status and ready for you to start experimenting with. Expect frequent changes as we add features and iterate based on feedback.

## Send a double ratchet message

Use the CLI to send a [double ratchet message](#double-ratchet-messaging) between test wallets using a local XMTP node and database running in Docker Desktop.

1. In the root directory of this repo, run:

   ```bash
   dev/up
   ```

> 2. Create a sender wallet account (user1). Create an [installation key bundle](#installation-key-bundles) and store it in the database. Grant the installation key bundle permission to message on behalf of the sender address. This will allow the CLI to message on behalf of the sender address.

Reviewer 2: Sorry, I thought this part of the document was going to be in the `examples/cli` folder. We should explicitly tell people to move to that folder between step 1 and 2, since the commands from 2 onwards will only work in that folder.

Author: Such a great callout @neekolas - thank you! The README is in `examples/cli`, but the `dev/up` command needs to be run at the root. So absolutely important to make going to `examples/cli` an explicit step.

Reviewer: We no longer need to run `dev/up`, because the CLI points to the dev network by default, which allows messaging across multiple machines.

If people explicitly want to run against localhost, they can run `dev/up` and then specify the `--local` flag on the CLI.

Author: Thank you @richardhuaaa! Updating now..... <3

   ```bash
   ./xli.sh --db user1.db3 register
   ```

3. Likewise, create a recipient wallet account (user2) and an installation key bundle.

   ```bash
   ./xli.sh --db user2.db3 register
   ```

4. Get the recipient's wallet address.

   ```bash
   ./xli.sh --db user2.db3 info
   ```

5. Send a message into the conversation. The message is sent using [one session between each installation](#installation-key-bundles) associated with the sender and recipient. The message is encrypted using a session key dedicated to this single message per the [double ratchet algorithm](#double-ratchet-messaging).

   ```bash
   ./xli.sh --db user1.db3 send <user2_address> "hello"
   ```

6. List conversations.

   ```bash
   ./xli.sh --db user1.db3 list-conversations

>    ```

Reviewer: @neekolas Should we add the upcoming list-messages command here as well?

Reviewer 2: We can do it as part of that PR

   ```bash
   dev/up
   ```

2. Change directories to `examples/cli`.

   ```bash
   cd examples/cli
   ```

3. Create a sender wallet account (user1). Create an [installation key bundle](#installation-key-bundles) and store it in the database. Grant the installation key bundle permission to message on behalf of the sender address. This will allow the CLI to message on behalf of the sender address.

   ```bash
   ./xli.sh --db user1.db3 register
   ```

3. Likewise, create a recipient wallet account (user2) and an installation key bundle.

   ```bash
   ./xli.sh --db user2.db3 register
   ```

4. Get the recipient's wallet address.

   ```bash
   ./xli.sh --db user2.db3 info
   ```

5. Send a message into the conversation. The message is sent using [one session between each installation](#installation-key-bundles) associated with the sender and recipient. The message is encrypted using a session key dedicated to this single message per the [double ratchet algorithm](#double-ratchet-messaging).

   ```bash
   ./xli.sh --db user1.db3 send <user2_address> "hello"
   ```

6. List conversations.

   ```bash
   ./xli.sh --db user1.db3 list-conversations
   ```

> ## Double ratchet messaging

Reviewer 3: This part (to the end) could probably be in the top level README. Doesn't seem specific to the CLI example.
